plugins/modules/win_domain_ou.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_domain_ou
+short_description: Manage Active Directory Organizational Units
+author: Joe Zollo (@joezollo)
+requirements:
+  - This module requires Windows Server 2012 or Newer
+description:
+  - Manage Active Directory Organizational Units
+  - Adds, Removes and Modifies Active Directory Organizational Units
+  - Task should be delegated to a Windows Active Directory Domain Controller
+options:
+  name:
+    description:
+      - The name of the Organizational Unit
+    type: str
+    required: true
+  protected:
+    description:
+      - Indicates whether to prevent the object from being deleted. When this
+        l(protected=true), you cannot delete the corresponding object without
+        changing the value of the property.
+    type: bool
+    default: false
+  path:
+    description:
+      - Specifies the X.500 path of the OU or container where the new object is
+        created.
+    type: str
+    required: true
+  state:
+    description:
+      - Specifies the desired state of the OU.
+      - When l(state=present) the module will attempt to create the specified
+        OU if it does not already exist.
+      - When l(state=absent), the module will remove the specified OU.
+      - When l(state=absent) and l(recursive=true), the module will remove all
+        the OU and all child OU's.
+    type: str
+    default: present
+    choices: [ present, absent ]
+  recursive:
+    description:
+      - Removes the OU and any child items it contains.
+      - You must specify this parameter to remove an OU that is not empty.
+    type: bool
+    default: false
+  properties:
```

Comment:
Should be a custom list of properties we know off or just directly mapped to the LDAP property name. We do the latter for some other domain modules in Ansible and it makes it more flexible in the future.

Reply:
Agreed! Will get this implemented!